src/cmd-cloud-prune

```diff
@@ -0,0 +1,300 @@
+                        prune_cloud_uploads(current_build, cloud_config, args.dry_run)
+                    case "build":
+                        raise NotImplementedError
+                        # print(f"Deleting key {prefix}{build.id} from bucket {bucket}")
+                        # Delete the build's directory in S3
+                        # S3().delete_object(args.bucket, f"{args.prefix}{str(current_build.id)}")
+                    case images:
+                        raise NotImplementedError
+            if not args.dry_run:
+                build.setdefault("policy-cleanup", []).append("cloud-uploads")
+                builds_json_data["builds"][index] = build
+                builds_json_data["timestamp"] = datetime.datetime.now(pytz.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    if not args.dry_run:
+        # Save the updated builds.json to local builds/builds.json
+        save_builds_json(builds_json_data)
+
+
+def get_s3_bucket_and_prefix(builds_source_data_url):
+    parsed_url = urlparse(builds_source_data_url)
+    if parsed_url.scheme == "s3":
+        bucket, prefix = parsed_url.netloc, parsed_url.path.lstrip("/")
+        return bucket, prefix
+    else:
+        raise Exception("Invalid scheme: only s3:// supported")
+
+
+def get_cloud_config(args):
+    return {
+        "gcp": {
+            "json-key": args.gcp_json_key,
+            "project": args.gcp_project
+        },
+        "aws": {
+            "credentials": args.aws_config_file
+        }
+    }
+
+
+def validate_policy(stream, policy):
+    # If the build key is set in the policy file, then the cloud-uploads key must
+    # also be present, and the duration of cloud-uploads must be equal or shorter
+    if "build" in policy[stream]:
+        actions = policy[stream]
+        if 'cloud-uploads' not in actions:
+            raise Exception("Pruning for cloud-uploads must be set before we prune the builds")
+        cloud_uploads_duration = get_period_in_months(actions["cloud-uploads"])
+        build_duration = get_period_in_months(actions["build"])
+        if cloud_uploads_duration > build_duration:
+            raise Exception("Duration of pruning cloud-uploads must be less than or equal to pruning a build")
+
+
+def get_supported_images(meta_json, unsupported, supported):
+    images = {}
+    for key in meta_json:
+        if key in unsupported:
+            raise Exception(f"The platform {key} is not supported")
+        if key in supported:
+            images[key] = meta_json[key]
+    return images
+
+
+def parse_fcos_version(version):
+    m = re.match(r'^([0-9]{2})\.([0-9]{8})\.([0-9]+)\.([0-9]+)$', version)
+    if m is None:
+        raise Exception(f"Incorrect versioning for FCOS build {version}")
+    try:
+        timestamp = datetime.datetime.strptime(m.group(2), '%Y%m%d')
+    except ValueError:
+        raise Exception(f"FCOS build {version} has incorrect date format. It should be in (%Y%m%d)")
+    return timestamp
```

Comment:
This can be moved to cosalib to be shared with versionary and the containers pruning code

Reply:
I was OK duplicating once but indeed now with the containers pruning code needing the same thing, it does make sense to put this in the library.

Not a blocker. We can do the move to a library in a follow-up or in the containers pruning PR.